CONTRIBUTING.md

```diff
@@ -0,0 +1,117 @@
+and a good way to start contributing to the project.
+
+It is essential that you know how to use Git as well as GitHub. While the maintainers will be more than happy helping
+you make your first PR, prerequisite knowhow makes it easier for everyone involved. Check this 
+[guide](https://github.com/firstcontributions/first-contributions)  if you've never contributed to a project on 
+GitHub before.
+
+Please note we have a [Code of Conduct](code-of-conduct.md). Please follow it in all your interactions with the project.
+
+When you submit code changes, your submissions are understood to be under the same [MIT License](LICENSE) that covers the 
+project. Feel free to contact the maintainers if that's a concern.
+
+## Moodle Endpoints Used By The App
+
+The Moodle endpoints used by this app have been collected in a [Postman](https://getpostman.com") workspace 
+[Here](https://www.getpostman.com/collections/e2c0439f144f7d3f60ed). You may import this collection into Postman directly using the link. 
+
+## Contributors Pull Request Checklist
+
+1. Fork the repository and branch from `development`.
+
+2. Work on your changes. Make sure you commit as frequently as required. Once you're done making the changes, push the commits
+   to your fork. Checkout the **Commit Messages** section to find guidelines relating to commit messages.
+
+3. If the changes you make fixes a bug in `master` or adds a new feature, note it down in the [changelog](CHANGELOD.md).
+   
+3. Head over to the [pull requests](https://github.com/crux-bphc/CMS-Android/pulls) page and create a new pull request. Make sure that the PR
+   is from your feature branch into this repo's `development` branch. If you are making changes that affect the UI in anyway, attach an image
+   in your PR description.
+
+4. Wait for a mainter to review your commit. If they finds any issue (including redundant code, unnecessary changes, potential bugs),
+   they will point it out to you. Make any necessary changes and push them to your fork. Ideally, try not to force push as it becomes
+   harded for the maintiners to keep track of changes. Mark the review as resolved.
+
+5. If you are a maintainer yourself, try to get a secondary opinion from other maintainers or contributors.
+
+6. If everything looks okay, the mainter(s) will go ahead and merge your changes into the repo. 
+
+**Note for maintainers**: When merging a PR in, *rebase the changes instead of creating a merge commit*. This ensures a clean commit history.
+Squash and rebase if required. You may also ask the contributor to do this if you are unsure how best to squash the commits. Changes introduced
+by a commit must be granular so that diagnosing and fixing problems later on is as simple as rolling back a specific commit.
+
+
+## Commit Messages
+
+Please check [this](https://chris.beams.io/posts/git-commit/) blog post on how to structure your commit messages. It is also a good idea to look
```

Comment:
While that is a good article, when I'm contributing to a project I want simple and succinct rules. See the fest-manager repo for an example of commit guielines

https://github.com/crux-bphc/fest-manager#commit-message-guidelines

Reply:
Thanks. I was too lazy to write the rules out myself. (Not to mention, I don't usually follow them myself). I'll rip out the one you've linked.

Reply:
Pushed new changes. Please go through them. Marking this as resolved. Thanks.